src/main/java/yanagishima/controller/ShareController.java

```diff
@@ -93,4 +126,55 @@ public void download(@RequestParam(name = "publish_id", required = false) String
     }
     return body;
   }
+
+  @PostMapping("share/updateViewers")
+  public Map<String, Object> updateViewers(@RequestParam(name = "publish_id") String publishId,
+                                           @RequestParam String viewers,
+                                           User user,
+                                           HttpServletResponse response) {
+    Map<String, Object> body = new HashMap<>();
+    try {
+      publishService.get(publishId).ifPresent(publish -> {
+        String publishUser = publish.getUser();
+        String requestUser = user.getId();
+        if (publishUser != null && publishUser.equals(requestUser)) {
+          publishService.update(publish, viewers);
+          body.put("viewers", viewers);
+          emitPublishEvent(publish.getPublishId(), publish.getDatasource(), publish.getEngine(),
+                  publish.getQueryId(), publish.getUser(), viewers);
+        } else {
+          AccessControlUtil.sendForbiddenError(response);
+          return;
+        }
+      });
+    } catch (Throwable e) {
+      log.error(e.getMessage(), e);
+      body.put("error", e.getMessage());
+    }
+    return body;
+  }
+
+  private void emitPublishEvent(String publishId, String datasource, String engine, String queryId, String user,
+                                String viewers) {
```

Comment:
refactor and add path because I guess API endpoint information is useful
https://github.com/yanagishima/yanagishima/pull/305/commits/00d7028fadeef757f51ee8b3d1ff2bb40ca9092c